core/src/main/scala/fs2/job/JobManager.scala

```diff
@@ -161,19 +178,18 @@ final class JobManager[F[_]: Concurrent: Timer, I, N] private (
   private[this] def managementMachinery[A](
       id: I,
       in: Stream[F, A],
-      startingTime: Timestamp,
-      ignoreAbsence: Boolean): Stream[F, A] = {
-    Stream force {
-      SignallingRef[F, Boolean](false) map { s =>
-        val unregisterJob =
-          Concurrent[F].delay(meta.remove(id)).void
-
-        // None indicates the job should not be started, otherwise returns the
-        // context expected prior to termination
-        lazy val frontF: F[Option[Context[F]]] =
+      isCanceled: SignallingRef[F, Boolean],
+      ignoreAbsence: Boolean)
+      : Stream[F, A] =
+    Stream.force(epochMillisNow map { startedAt =>
+      // Results in the context expected prior to termination, `None` indicates
+      // the job should not be started.
+      lazy val attemptF: F[Option[Context[F]]] = {
+        val startF: F[Option[Context[F]]] =
           Concurrent[F].delay(Option(meta.get(id))) flatMap {
-            case Some(old @ Context(Status.Pending, _)) =>
-              val running = Context(Status.Running, Some(s.set(true)))
+            // `submit`ted job
+            case Some(old @ Context(Status.Pending, cancel)) =>
+              val running = Context(Status.Running, cancel)
 
               val casF = Concurrent[F] delay {
```

Comment:
Wasn't changed in this PR but maybe rename `casF` to `isReplacedF` or something? (and something similar with the `casF` occurrence below).

Also not changed in this PR, but I was thinking that we may simplify things here by getting rid of the `attemptF` calling itself, if we change the `meta.get` / `meta.replace` / `meta.putIfAbsent` in 1 go with e.g. `meta.merge`.

Reply:
I'll PR some more descriptive names 👍 

As for `meta.merge`, I'm not convinced that will simplify things as we recurse on `attemptF` in part to perform the cancellation check again.

Reply:
> we recurse on attemptF in part to perform the cancellation check again

Ok but isn't that check needed precisely because it isn't read and written in 1 go? I mean, with `merge` I imagine 

```
    replacedF flatMap { replaced =>         <-- we know `replaced` is true
      if (replaced)
        Concurrent[F].pure(Some(running))   <-- ... so only have to handle this
      else
        attemptF                            <-- ... not this
    }
```

and similar for `putF`

Reply:
I'll merge anyway. Wasn't meant to be changed right now, just as something we could possibly improve whenever there's an opportunity.

Reply:
> Ok but isn't that check needed precisely because it isn't read and written in 1 go?

The cancelation state is no longer kept in the `meta` map (which allows us to have the map only reflect active jobs and avoid the sentinel state), it requires reading [`isCanceled`](https://github.com/wemrysi/fs2-job/blob/1c560e2ae1b04ee2f0853aa15c2aa1ccf94e1ca8/core/src/main/scala/fs2/job/JobManager.scala#L227) again.